### PR TITLE
Migration guide for v2

### PR DESCRIPTION
**Pull request by jwmueller**
Added migration guide for existing users to docs.

Note there is link to yet-unreleased Release Notes, it's probably the right link but needs to be verified. 
Link will also not work until the 2.0.0 release is tagged.

**Squash and Merge this PR**

### CODE REVIEW

File: docs/source/migrating/migrate_v2.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+
+Method and class name changes
+-----------------------------
+
+| **Old:** ``pruning.get_noise_indices(s, psx, prune_method, sorted_index_method, ...)``
+| --> 
+| **New:** ``filter.find_label_issues(labels, pred_probs, filter_by, return_indices_ranked_by, ...)``
+
+**Note:** ``inverse_noise_matrix`` is no longer a supported input argument, but ``confident_joint`` remains (you can easily convert between these two).
```
Comment (anishathalye) [inline docs/source/migrating/migrate_v2.rst:17]: Convert how? Can we link them to a function?

File: docs/source/migrating/migrate_v2.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+Below we outline the major updates and code substitutions to be aware of. 
+The full change-log is listed in the `v2.0.0. Release Notes <https://github.com/cleanlab/cleanlab/releases/tag/v2.0.0>`_.
+
+
+Method and class name changes
+-----------------------------
+
+| **Old:** ``pruning.get_noise_indices(s, psx, prune_method, sorted_index_method, ...)``
+| --> 
+| **New:** ``filter.find_label_issues(labels, pred_probs, filter_by, return_indices_ranked_by, ...)``
```
Comment (anishathalye) [inline docs/source/migrating/migrate_v2.rst:15]: Can we cross-ref all the new class/function/method names, so they are clickable?

File: docs/source/migrating/migrate_v2.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+| -->
+| **New:** ``classification.CleanLearning(..., find_label_issues_kwargs)``
+
+**Note:** ``CleanLearning`` can now find label errors for you, neatly organizing them in a pandas.DataFrame as well as computing the required out-of-sample predicted probabilities. You just specify which classifier, we handle the cross-validation!
+
+
+Module name changes
+-------------------
+
+``pruning.py`` --> ``filter.py``
```
Comment (anishathalye) [inline docs/source/migrating/migrate_v2.rst:42]: Can cross-ref modules as well, with `:py:mod:`

**Review by anishathalye — COMMENTED** [review]
Note: remember to either squash and merge, or clean up the history before merging this in.

**Comment by anishathalye** [comment]
The "Migrating to v2.0.0" shows up in the "Links" section on the sidebar. Is that intentional?

File: docs/source/index.rst [hunk 4]
```diff
@@ -108,3 +108,10 @@ When the :py:meth:`.fit() <cleanlab.classification.CleanLearning.fit>` method is
    GitHub <https://github.com/cleanlab/cleanlab>
    PyPI <https://pypi.org/project/cleanlab/>
    Conda <https://anaconda.org/Cleanlab/cleanlab>
+
+.. toctree::
+   :hidden:
+
+   Migrating to v2.0.0 <migrating/migrate_v2>
```
Comment (jwmueller) [inline docs/source/index.rst:115]: > "Migrating to v2.0.0" shows up in the "Links" section on the sidebar. Is that intentional?

@anishathalye Yes that's where I put it, feel free to move it elsewhere if you have a better spot.

Comment (anishathalye) [reply docs/source/index.rst:115]: Seems fine for now, we can consider moving it later.

**Review by anishathalye — APPROVED** [review]

**Pull request merged**